modal/_container_entrypoint.py

```diff
@@ -441,20 +553,110 @@ def import_function(
-        function_def.is_auto_snapshot,
-        function,
+        code_deps,
+    )
+
+
+def import_class_service(
+    function_def: api_pb2.Function,
+    ser_cls,
+    ser_fun,
+    ser_params: Optional[bytes],
+    client: Client,
+) -> Service:
+    """
+    This imports a full class to be able to execute any @method or webhook decorated methods.
+
+    See import_function.
+    """
+    module: Optional[ModuleType] = None
+    cls: Optional[Type] = None
+    active_app: Optional[_App] = None
+    code_deps: Optional[List["modal.object._Object"]] = None
+
+    if function_def.definition_type == api_pb2.Function.DEFINITION_TYPE_SERIALIZED:
+        assert ser_cls is not None
+        cls = ser_cls
+        # This is a serialized function we already fetched from the server
+        method_partials = _find_partial_methods_for_cls(cls, _PartialFunctionFlags.all())
+    else:
+        # Load the module dynamically
+        module = importlib.import_module(function_def.module_name)
+        qual_name: str = function_def.function_name
+
+        if not is_global_object(qual_name):
+            raise LocalFunctionError("Attempted to load a class defined in a function scope")
+
+        parts = qual_name.split(".")
+        if not (
+            len(parts) == 2 and parts[1] == "*"
+        ):  # the "function name" of a class service "function placeholder" is expected to be "ClassName.*"
+            raise InvalidError(f"Invalid 'class function' identifier {qual_name}")
+
+        assert not function_def.use_method_name  # new "placeholder methods" should not be invoked directly!
+        cls_name = parts[0]
+        cls = getattr(module, cls_name)
+        if isinstance(cls, Cls):
+            # The cls decorator is in global scope
+            _cls: "modal.cls._Cls" = synchronizer._translate_in(cls)
+            method_partials = _cls._method_partials  # implementation to follow in cls refactor PR
+            active_app = _cls._app
+            # TODO: set code deps using the decorator options of the cls - maybe using _cls.deps?
+        else:
+            # This is a raw class - find all methods
+            method_partials = _find_partial_methods_for_cls(cls, _PartialFunctionFlags.all())
+
+    # Instantiate the class if it's defined
+    assert cls  # must be a class
+    if ser_params:
+        _client: _Client = synchronizer._translate_in(client)
+        args, kwargs = deserialize(ser_params, _client)
+    else:
+        args, kwargs = (), {}
+    obj = cls(*args, **kwargs)
+    if isinstance(cls, Cls):
+        obj = obj.get_obj()
+
+    return ImportedClass(
+        obj,
+        method_partials,
+        active_app,
+        code_deps,
     )
 
 
+def get_active_app_fallback(function_def: api_pb2.Function) -> Optional[_App]:
+    # This branch is reached in the special case that the imported function/class is:
+    # 1) not serialized, and
+    # 2) isn't a FunctionHandle - i.e, not decorated at definition time
```

Comment:
Do we still have `FunctionHandle` as a concept?

Reply:
This is an old comment that was moved here I think. We don't have separate Function/FunctionHandle concepts, but this refers to it being a proper `modal.Function` rather than just a Python vanilla function (i.e. decorated).
This handles the rare case, mostly used in tests, that the imported raw function isn't actually decorated in global scope and is a vanilla Python function that doesn't know which Modal app it belongs to.